src/Faker/Provider/en_GB/Company.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Faker\Provider\en_GB;
+
+class Company extends \Faker\Provider\Company
+{
+    public const VAT_PREFIX = 'GB';
+    public const VAT_TYPE_DEFAULT = 'vat';
+    public const VAT_TYPE_BRANCH = 'branch';
+    public const VAT_TYPE_GOVERNMENT = 'gov';
+    public const VAT_TYPE_HEALTH_AUTHORITY = 'health';
+
+    /**
+     * UK VAT number
+     *
+     * @see https://en.wikipedia.org/wiki/VAT_identification_number#VAT_numbers_by_country
+     */
+    public static function vat(string $type = null): string
+    {
+        switch ($type) {
+            /**
+             * branch traders:
+             * 12 digits (as for 9 digits, followed by a block of 3 digits)
+             */
+            case static::VAT_TYPE_BRANCH:
+                return sprintf(
+                    '%s%d %d %d %d',
+                    static::VAT_PREFIX,
+                    static::randomNumber(3, true),
```

Comment:
This only produces numbers bigger than 100. Is that on purpose? If not you could use the `str_pad` solution from below here too. The other cases that use `randomNumber` have the same issue.

Reply:
Personally I'd rather stick with the simpler `randomNumber()` generator where possible. If I was wanting to test the behaviour of numbers with leading zeros I would use a defined value rather than faking it.

Reply:
You could also use `static::numerify('###')`.